spec/helpers/gov_one_helper_spec.rb

```diff
@@ -4,28 +4,36 @@
   describe '#login_uri' do
     subject(:login_uri) { helper.login_uri }
 
-    it 'returns a URI object with the correct host and path' do
-      expect(login_uri.host).to eq 'oidc.test.account.gov.uk'
-      expect(login_uri.path).to eq '/authorize'
-    end
-
     it 'encodes the authorize endpoint params' do
-      expect(login_uri.query).to start_with 'response_type=code&scope=email+openid&client_id='
-      expect(login_uri.query).to end_with '&redirect_uri=http%3A%2F%2Frecovery.app%2Fusers%2Fauth%2Fopenid_connect%2Fcallback'
+      expect(login_uri).to start_with 'https://oidc.test.account.gov.uk/authorize?response_type=code&scope=email+openid&client_id='
+      expect(login_uri).to end_with 'redirect_uri=http%3A%2F%2Frecovery.app%2Fusers%2Fauth%2Fopenid_connect%2Fcallback'
     end
   end
 
   describe '#logout_uri' do
     subject(:logout_uri) { helper.logout_uri }
 
-    it 'returns a URI object with the correct host and path' do
-      expect(logout_uri.host).to eq 'oidc.test.account.gov.uk'
-      expect(logout_uri.path).to eq '/logout'
+    it 'encodes the logout endpoint params' do
```

Comment:
If you preferred having a URI object at any point you can move `to_s` into the button helpers.

Reply:
@peterdavidhamilton I've updated the helper, was having trouble getting the login link to work - issue seems to be fixed by not encoding the redirect param

Reply:
Is that in local dev or when testing?